xtask/codegen/css.ungram

```diff
@@ -545,6 +567,35 @@ CssAllProperty =
 // recovery and representing "valid but incorrect" inputs.
 AnyCssAllPropertyValue = CssWideKeyword
 
+/// https://drafts.csswg.org/css-backgrounds/#propdef-border
+/// border =
+///   <line-width> ||
+///   <line-style> ||
+///   <color>
+CssBorderProperty =
+	name: CssIdentifier
+	':'
+	value: AnyCssBorderPropertyValue
+
+AnyCssBorderPropertyValue =
+	CssBorder
+
+CssBorder =
+	line_width: AnyCssLineWidth ||
+	line_style: CssLineStyle ||
+	color: CssColor
+
+// TODO: This should be narrowed down to CssLength when we can.
+AnyCssLineWidth =
+	CssRegularDimension
+	| CssLineWidthKeyword
+
+CssLineWidthKeyword = keyword: ('thin' | 'medium' | 'thick')
+
+CssLineStyle =
+	keyword: ( 'none' | 'hidden' | 'dotted' | 'dashed' | 'solid' | 'double'  | 'groove' | 'ridge' | 'inset' | 'outset')
```

Comment:
I think it would be great if we could just make this a generic "CssKeywordValue" and have the parser enforce what the value is, otherwise we're going to end up with 100 of these `Css*Keyword` node types and I think that's not really all that helpful.

I would rather be able to write this whole thing as:

```
CssBorder =
  width: CssLineWidth || ...
  
AnyCssLineWidth =
  CssRegularDimension
  | keyword: ('thin' | 'medium' | 'thick')
```

But our current implementation of codegen won't understand that and just generates a regular node with 4 different `keyword_token` members lol. Thankfully that will mostly be an infrastructure change and I think we can safely handle that in the future.